examples/loki/nurf_data.py

```diff
@@ -102,120 +102,173 @@ def nurf_file_creator(loki_file, path_to_loki_file, data):
+       
+
+        # subgroup for fluo_integration_time
+        fluo_inttime_data = grp_fluo.create_dataset('fluo_integration_time',
+                                               data=data['Fluo_IntegrationTime'],
+                                               shape=data['Fluo_IntegrationTime'].shape,
                                                dtype=np.int32)
-        fluo_inttime.attrs['units'] = 's'  # TODO: unit to be verified
-        fluo_inttime.attrs['long name'] = 'fluo_integration_time'
-
-        fluo_bkg = grp_fluo.create_dataset('fluo_background',
-                                           data=data['Fluo_background'],
-                                           shape=data['Fluo_background'].shape,
-                                           dtype=np.float32)
-        fluo_bkg.attrs['long name'] = 'fluo_background'
-
-        fluo_int0 = grp_fluo.create_dataset('fluo_intensity0',
-                                            data=data['Fluo_intensity0'],
-                                            shape=data['Fluo_intensity0'].shape,
-                                            dtype=np.float32)
-        fluo_int0.attrs['long name'] = 'fluo_intensity0'
-
-        fluo_monowavelengths = grp_fluo.create_dataset('fluo_monowavelengths',
+        fluo_inttime_data.attrs['units'] = 'us'  # TODO: unit to be verified, currently micro-seconds
+        fluo_inttime_data.attrs['long name'] = 'fluo_integration_time'
+
+
+        # subgroup for fluo_monowavelengths
+        fluo_monowavelengths_data = grp_fluo.create_dataset('fluo_monowavelengths',
                                                        data=data[
                                                            'Fluo_monowavelengths'],
                                                        shape=data[
                                                            'Fluo_monowavelengths'].shape,
                                                        dtype=np.float32)
-        fluo_monowavelengths.attrs['units'] = 'nm'  # TODO: unit to be verified
-        fluo_monowavelengths.attrs['long name'] = 'fluo_monowavelengths'
+        fluo_monowavelengths_data.attrs['units'] = 'nm'  # TODO: unit to be verified
+        fluo_monowavelengths_data.attrs['long name'] = 'fluo_monowavelengths'
 
-        fluo_spectra = grp_fluo.create_dataset('fluo_spectra',
-                                               data=data['Fluo_spectra'],
-                                               shape=data['Fluo_spectra'].shape,
-                                               dtype=np.float32)
-        fluo_spectra.attrs['long name'] = 'fluo_spectra'
-
-        fluo_wavelength = grp_fluo.create_dataset('fluo_wavelength',
+        
+        # subgroup for fluo_wavelength
+        fluo_wavelength_data= grp_fluo.create_dataset('fluo_wavelength',
                                                   data=data['Fluo_wavelength'],
                                                   shape=data[
```

Comment:
As above: Can't you omit the shape?

Reply:
Yes, I can, but I learned earlier it is better to provide the shape. But maybe that was wrong?

Reply:
I implemented most of your suggestions, but I am not sure if I should execute a new pull request. https://github.com/RGBYCP/generate-nexus-files/commit/d90161805bfb0b0adc63fb30683e0d854cd11fe3
Not yet done `auxiliary_signals` and I left wavelength, time as own datasets.

Reply:
> but I am not sure if I should execute a new pull request

No new PR, just push to this branch with your updates.